File: frontend/src/app/shared/dto/message.dto.ts
export class MessageDto {
  public readonly id: number | null;
  public readonly timestamp: Date;
  public readonly body: string;
  public readonly status: string;
  public readonly labels: string;
  public readonly senderId: number;
  public readonly receiverId: number;

  constructor(
    id: number | null,
    timestamp: Date,
    body: string,
    status: string,
    labels: string,
    senderId: number,
    receiverId: number
  ) {
    this.id = id;
    this.timestamp = timestamp;
    this.body = body;
    this.status = status;
    this.labels = labels;
    this.senderId = senderId;
    this.receiverId = receiverId;
  }
}

File: backend/src/main/java/com/example/backend/model/dto/MessageDTO.java
package com.example.backend.model.dto;

import java.util.Date;

public class MessageDTO {
    private Long id;
//    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
    private Date timestamp;
    private String body;
    private String status;
    private String labels;
    private Long senderId;
    private Long receiverId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Date getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Date timestamp) {
        this.timestamp = timestamp;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getLabels() {
        return labels;
    }

    public void setLabels(String labels) {
        this.labels = labels;
    }

    public Long getSenderId() {
        return senderId;
    }

    public void setSenderId(Long senderId) {
        this.senderId = senderId;
    }

    public Long getReceiverId() {
        return receiverId;
    }

    public void setReceiverId(Long receiverId) {
        this.receiverId = receiverId;
    }
}
